perfkitbenchmarker/providers/openstack/os_disk.py

```diff
@@ -95,6 +95,13 @@ def Attach(self, vm):
                 is_unattached = not(volume.status == "in-use"
                                     and volume.attachments)
 
+        for attachment in volume.attachments:
```

Comment:
Well, at this point since the while loop above never times out until there is an attachment, it will always have a matching attachment.

Perhaps, I could improve it by making it timeout, and throwing an appropriate error message.